drafts/2.4-representable-functors.md

```diff
@@ -35,50 +39,62 @@ instance Profunctor (->) where
     rmap = (.)
 ```
 ```kotlin:ank:silent
-
+import arrow.extension
+import arrow.typeclasses.Profunctor
+@extension
+interface ArrowProfunctor: Profunctor<ForFunction1> {
+    override fun <A, B, C, D> Kind2<ForFunction1, A, B>.dimap(fl: (C) -> A, fr: (B) -> D): Kind2<ForFunction1, C, D> =
+        (fr compose fix().f compose fl).k()
+ 
+    override fun <A, B, C> Kind2<ForFunction1, A, B>.lmap(f: (C) -> A): Kind2<ForFunction1, C, B> =
+        (f andThen fix().f).k()
+ 
+    override fun <A, B, D> Kind2<ForFunction1, A, B>.rmap(f: (B) -> D): Kind2<ForFunction1, A, D> =
+        (fix().f andThen f).k()
+}
 ```
 ................
 ```Haskell
 alpha :: forall x. (a -> x) -> F x
 ```
-```kotlin:ank:silent
-
+```kotlin:kotlin
+fun<A, F, G> alpha(): (Kind<F, A>) -> Kind<G, A>
 ```
 ................
 ```Haskell
 fmap f . alpha = alpha . fmap f
 ```
-```kotlin:ank:silent
-
+```kotlin
+fmap(f) compose alpha == alpha compose fmap(f)
 ```
 ................
 ```Haskell
 fmap f (alpha h) = alpha (f . h)
 ```
-```kotlin:ank:silent
-
+```kotlin
+fmap(f)(alpha(h)) == alpha(f compose h)
 ```
 ................
 ```Haskell
 beta :: forall x. F x -> (a -> x)
 ```
 ```kotlin:ank:silent
-
+fun<A, F, G> beta(): (Kind<F, A>) -> Kind<G, A>
 ```
 ................
 ```Haskell
 alpha :: forall x. (Int -> x) -> [x]
 alpha h = map h [12]
 ```
-```kotlin:ank:silent
-
+```kotlin
+fun <A> alpha(h : (Int) -> A) = listOf(12).map(h)
 ```
 ................
 ```Haskell
 map f (map h [12]) = map (f . h) [12]
 ```
-```kotlin:ank:silent
-
+```kotlin
+fmap(f)(fmap(h)(listOf(12))) == fmap(f compose h)(listOf(12))
```

Comment:
These would be `map` functions not `fmap`, because we assume the functions `h` and `f`  go from some `(Y) -> Z` and we stay in the List container.
Both ways work, but map fits more to the haskell code

Reply:
should I change the fmap name to map?
```Haskell
instance Functor (Reader a) where
    fmap f h = f . h
```
```kotlin:ank:silent
fun <D, A> fmap(run: (D) -> A): Reader<D, A> = Kleisli(run.andThen { Id(it) })
```

Reply:
Come again?